src/openrct2/drawing/Drawing.h

```diff
@@ -108,29 +108,80 @@ struct Gx
 struct DrawPixelInfo
 {
     uint8_t* bits{};
+
+private:
     int32_t x{};
     int32_t y{};
     int32_t width{};
     int32_t height{};
+
+public:
     int32_t pitch{}; // note: this is actually (pitch - width)
     ZoomLevel zoom_level{};
 
-    /**
-     * As x and y are based on 1:1 units, zooming in will cause a reduction in precision when mapping zoomed-in
-     * pixels to 1:1 pixels. When x, y are not a multiple of the zoom level, the remainder will be non-zero.
-     * The drawing of sprites will need to be offset by this amount.
-     */
-    uint8_t remX{};
-    uint8_t remY{};
-
     // Last position of drawn text.
     ScreenCoordsXY lastStringPos{};
 
     OpenRCT2::Drawing::IDrawingEngine* DrawingEngine{};
 
-    size_t GetBytesPerRow() const;
     uint8_t* GetBitsOffset(const ScreenCoordsXY& pos) const;
     DrawPixelInfo Crop(const ScreenCoordsXY& pos, const ScreenSize& size) const;
+
+    constexpr int32_t WorldX() const
+    {
+        return zoom_level.ApplyTo(x);
+    }
+    constexpr int32_t WorldY() const
+    {
+        return zoom_level.ApplyTo(y);
+    }
+    constexpr int32_t WorldWidth() const
+    {
+        return zoom_level.ApplyTo(width);
+    }
+    constexpr int32_t WorldHeight() const
+    {
+        return zoom_level.ApplyTo(height);
+    }
+
+    constexpr int32_t ScreenX() const
+    {
+        return x;
+    }
+    constexpr int32_t ScreenY() const
+    {
+        return y;
+    }
+    constexpr int32_t ScreenWidth() const
+    {
+        return width;
+    }
+    constexpr int32_t ScreenHeight() const
+    {
+        return height;
+    }
```

Comment:
I have to disagree with those getters/setters, I don't really see the point when its just a lot easier to access the members of a struct, I would argue only use getters/setters when it typically involves side effects, otherwise this makes the code is very noisy and makes debug builds typically even slower as no inlining will take place causing a lot of branching. Probably best to just rename the variables so they are more appropriate like the getter/setter names. Utility functions that transform the state and result something is okay.

Reply:
I agree, I just set them up for the refactor. Removed.